# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.h
# Commit: 2fc9a3d0ad8b
# Full Hash: 2fc9a3d0ad8bdbce325df78fcd731e925245afaf
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-12 09:23:07
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 1 - CookieStore API - WebIDL, r=smaug,edgul
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215144
# ==============================================================================

diff -r 2d598be5ec67 -r 2fc9a3d0ad8b dom/workers/WorkerScope.h
--- a/dom/workers/WorkerScope.h	Wed Sep 11 16:09:37 2024 +0000
+++ b/dom/workers/WorkerScope.h	Wed Sep 11 16:26:54 2024 +0000
@@ -55,6 +55,7 @@
 class ClientSource;
 class Clients;
 class Console;
+class CookieStore;
 class Crypto;
 class DOMString;
 class DebuggerNotificationManager;
@@ -482,6 +483,8 @@
 
   void EventListenerAdded(nsAtom* aType) override;
 
+  already_AddRefed<mozilla::dom::CookieStore> CookieStore();
+
   IMPL_EVENT_HANDLER(message)
   IMPL_EVENT_HANDLER(messageerror)
 
@@ -491,6 +494,8 @@
   IMPL_EVENT_HANDLER(push)
   IMPL_EVENT_HANDLER(pushsubscriptionchange)
 
+  IMPL_EVENT_HANDLER(cookiechange)
+
  private:
   ~ServiceWorkerGlobalScope();
 
@@ -500,6 +505,7 @@
   const nsString mScope;
   RefPtr<ServiceWorkerRegistration> mRegistration;
   SafeRefPtr<extensions::ExtensionBrowser> mExtensionBrowser;
+  RefPtr<mozilla::dom::CookieStore> mCookieStore;
 };
 
 class WorkerDebuggerGlobalScope final : public WorkerGlobalScopeBase {